kinesis_consumer.cpp

```diff
@@ -148,6 +149,28 @@ lookup_region(const char *r)
 	return -1;
 }
 
+static void
+parse_url(ClientConfiguration &config,
```

Comment:
No worries - They do have a built-it URI parser, it doesn't do any validation though. The URL gets handled down to curl and bubbles back up as a curl error. I concur that a more standard parse and validate would be nice though - TODO